src/Processor/ProductAttribute/AssetProductAttributeProcessor.php

```diff
@@ -0,0 +1,129 @@
+        string $assetFamilyCode,
+        string $assetCode,
+        string $attributeCode,
+        array $assetAttributeResource = [],
+    ): bool {
+        if (!$this->assetAttributeValueBuilder->hasSupportedBuilder($assetFamilyCode, $attributeCode)) {
+            return false;
+        }
+
+        if (!\method_exists($model, 'addAsset')) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function process(
+        ProductInterface $model,
+        string $assetFamilyCode,
+        string $assetCode,
+        string $attributeCode,
+        array $assetAttributeResource = [],
+    ): void {
+        if (!$this->support($model, $assetFamilyCode, $assetCode, $attributeCode)) {
+            return;
+        }
+
+        $scope = $this->productFilterRulesProvider->getProductFiltersRules()->getChannel();
+        $queryParam = [
+            'familyCode' => $assetFamilyCode,
+            'attributeCode' => $attributeCode,
+            'assetCode' => $assetCode,
+            'scope' => $scope,
+        ];
+
+        foreach ($this->syliusAkeneoLocaleCodeProvider->getUsedLocalesOnBothPlatforms() as $syliusLocale) {
+            $queryParam['locale'] = $syliusLocale;
+            $this->handleAsset($model, $assetFamilyCode, $assetCode, $attributeCode, $queryParam, $assetAttributeResource);
+        }
+    }
+
+    private function handleAsset(
+        ProductInterface $model,
+        string $assetFamilyCode,
+        string $assetCode,
+        string $attributeCode,
+        array $queryParam,
+        array $assetAttributeResource,
+    ): void {
+        $asset = $this->assetRepository->findOneBy($queryParam);
+
+        if (!$asset instanceof Asset) {
+            /** @var Asset $asset */
+            $asset = $this->assetFactory->createNew();
+            $this->entityManager->persist($asset);
+            $asset->setFamilyCode($assetFamilyCode);
+            $asset->setAssetCode($assetCode);
+            $asset->setAttributeCode($attributeCode);
+            $asset->setLocale($queryParam['locale']);
+            $asset->setScope($queryParam['scope']);
+            $asset->setType($this->akeneoAssetAttributePropertiesProvider->getType($assetFamilyCode, $attributeCode));
+        }
+
+        $akeneoLocale = $this->syliusAkeneoLocaleCodeProvider->getAkeneoLocale($queryParam['locale']);
+
+        $assetAttributeValue = $this->akeneoAssetAttributeDataProvider->getData(
+            $assetFamilyCode,
+            $attributeCode,
+            $assetAttributeResource,
+            $akeneoLocale,
+            $queryParam['scope'],
+        );
+
+        /** @var array $data */
+        $data = $this->assetAttributeValueBuilder->build(
+            $assetFamilyCode,
+            $attributeCode,
+            $akeneoLocale,
+            $queryParam['scope'],
+            $assetAttributeValue,
+        );
+
+        $asset->setContent($data);
+
+        if (!\method_exists($model, 'addAsset')) {
```

Comment:
I think, you can do it at the top of de function to awoid process on it.

Reply:
its already in the support function, it should always be true in the process, but its there to only add the asset on the product if method is available otherwise its only imported into the asset table